app/modules/net.c

```diff
@@ -763,11 +771,14 @@ int net_close( lua_State *L ) {
   }
   if (ud->type == TYPE_TCP_SERVER ||
      (ud->pcb == NULL && ud->client.wait_dns == 0)) {
-    lua_gc(L, LUA_GCSTOP, 0);
+//    lua_gc(L, LUA_GCSTOP, 0);
```

Comment:
I believe these are intentional, to guard against the GC kicking in on the unref of `ud`, rendering the `ud->self_ref = LU_NOREF;` as undefined behaviour due to a use-after-free scenario.

Reply:
The UD is still on the Lua stack so the `luaL_unref()` _cannot_ trigger its GC;  there is no point in stopping GC around this statement.

Reply:
Ah, of course, yes, you're right. Let's get rid of those (commented) calls altogether then, they're plain misleading and encouraging bad copy/pasta :)  I mean, they had me fooled...